Task/tailor for update templates - For review only

henrjk opened the pull request:
**For review only -- not quite ready to be merged yet**

This implements the changes described in comment https://github.com/opendevstack/ods-project-quickstarters/issues/38#issuecomment-499946943.

The code requires https://github.com/opendevstack/ods-jenkins-shared-library/pull/91 so that the openshift folder generated is actually being processed with tailor during the jenkins build. 

Besides this requirement the code is not ready to be pulled in as it currently has rundeck job copies for testing. This will have to be cleaned up before delivering.

https://github.com/opendevstack/ods-jenkins-shared-library/pull/91 fixes some problems encountered with https://github.com/opendevstack/ods-jenkins-shared-library/pull/85.

Please review!

gerardcl commented on boilerplates/be-node-express/Jenkinsfile:
```diff
@@ -9,18 +9,18 @@ node {
 }
 
-library identifier: 'ods-library@production', retriever: modernSCM(
+library identifier: 'ods-library@feature/tailor-deploy-debug', retriever: modernSCM(
```
@henrjk I would recommend the identifier to match this PR source branch name, or to point to production. Anyway, as you already mentioned, if the purpose of this PR is only for researching and proposing yet it is fine for me!

henrjk replied:
Yes, I wanted to remove this change (so that it would be production) once the fix (or something equivalent) in branch feature/tailor-deploy-debug is merged.

michaelsauter commented:
* Unify multiple yml into one
* Remove references to PR branch
* Ensure all params are present in configuration sample

henrjk commented:
I updated the PR with 
- Unify multiple yml into one
- Remove references to PR branch

This has been tested in production with the mentioned PR branch of the shared library. **It should be retested with the final version of the shared library**

Also before this can be merged (after additional review) I will still need to:
- Ensure all params are present in configuration sample. 

I will leave another comment when I am done.

I discussed the PR with @gerardcl and @hugowschneider. Based on their feedback I changed the way the openshift folder is generated so it is easier to customize. See comments in https://github.com/opendevstack/ods-project-quickstarters/blob/955f6c69679a2b68f864f4d5d8ee900d7a59c993/boilerplates/init-openshift.sh#L5

Other things we talked about which I do not want to pursue in this issue were:
- Could the openshift folder be used for migration and would bring advantages compared to the current approach with occonfig_artifacts?
- Since the delete-component.sh scripts have been removed, give instructions in the component creation output on how to delete the generated component based on the app label.
- Should tailor use the --upsert-only  switch when updating based on the openshift folder (perhaps only for the initial update)?
And some more stylistic changes:
- Combine several consecutive script executions in rundeck jobs into a single script. However we are not sure whether this work would carry forward once we move away from rundeck.
- Also use script instead of exec everywhere in the rundeck script. I would like to do this separately or postpone as it will require retesting all changed scripts.

henrjk commented:
I consider this now complete.
